File: Arrays/PascalTri/Go/118-Pascal's-Triangle.go
func generate(numRows int) [][]int {
        if numRows == 0 {
        return [][]int{}
    }

    res := [][]int{{1}}

    for i := 1; i < numRows; i++ {
        lastRow := res[len(res)-1]
        preRow := append([]int{0}, lastRow...)
        preRow = append(preRow, 0)

        row := []int{}
        for j := 0; j < len(lastRow)+1; j++ {
            row = append(row, preRow[j]+preRow[j+1])
        }
        res = append(res, row)
    }
    return res    
}

File: Arrays/PascalTri/Python/118-Pascal's-Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []

        res = [[1]]

        for i in range(1, numRows):
            preRow = res[-1]
            temp = [0] + preRow + [0]
            row = [temp[j] + temp[j+1] for j in range(len(preRow)+1)]
            res.append(row)
            
        return res
        